source/framefmt.c

```diff
@@ -213,6 +214,22 @@ static inline int serde_nonframed(
     return AWS_ERROR_SUCCESS;
 }
 
+bool aws_cryptosdk_frame_is_valid(const struct aws_cryptosdk_frame *const frame) {
+    bool iv_valid = aws_byte_buf_is_valid(&frame->iv);
+    bool ciphertext_valid = aws_byte_buf_is_valid(&frame->ciphertext);
+    /* This happens when input plaintext size is 0 */
+    bool ciphertext_valid_zero = frame->ciphertext.len == 0 &&
```

Comment:
A byte buf with capacity 0 in c-common is only considered valid if its buffer pointer is `NULL`. However `serialize` (when given plaintext_size = 0) can initialize the `ciphertext` byte buffer in the returning frame to point to a valid location in `ciphertext_buf` but to have 0 capacity. IMO it should be required that `plaintext_size > 0` in order to prevent this behaviour. Alternatively, we could relax the validity (as shown here), but I think that there is no reason to allow `plaintext_size == 0`.

Reply:
Plaintext of size 0 is explicitly allowed for processing by all the AWS Encryption SDK implementations and required in the feature spec.